Add support for using io.opentracing.tag Tags directly, support keywords

Limess opened the pull request:
This allows easier use of the convenients specified here https://github.com/opentracing/specification/blob/master/semantic_conventions.md by using the static variables here: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/tag/Tags.java.

Has the side-effect of supporting keywords directly on `set-tag` (this includes new test cases).

Additionally fixed some minor reflection warnings:

```
Reflection warning, opentracing_clj/core.clj:23:4 - reference to field scopeManager can't be resolved.
Reflection warning, opentracing_clj/core.clj:32:5 - reference to field activeSpan can't be resolved.
Reflection warning, opentracing_clj/core.clj:44:3 - call to method activate can't be resolved (target class is unknown).
```

Limess committed:
Add support for using io.opentracing.tag Tags directly, support keywords

This allows easier use of the convenients specified here https://github.com/opentracing/specification/blob/master/semantic_conventions.md.

Limess committed:
Fix reflection warning when fetching scope-manager

alvinfrancis commented on src/clj/opentracing_clj/core.clj:
```diff
@@ -117,16 +117,22 @@
-     :else                     (.setTag span key ^String (str value)))))
+     (instance? Boolean value) (.setTag span ^String (resolve-tag-key key) ^Boolean value)
+     (instance? Number value)  (.setTag span ^String (resolve-tag-key key) ^Number value)
+     (instance? Tag value)     (.setTag span ^String (resolve-tag-key key) ^String (.getKey ^Tag value))
```
As I understand it, an io.opentracing.tag cannot be the value for another io.opentracing.tag.  The constants used for a tag value in the Tags object are all non-tags.

alvinfrancis reviewed (changes requested):
Looks good overall save for one thing. :+1:

Limess committed:
Remove support for Tag instances as values

Correctly identified that the signature of generic method support for this is (Tag<T>, T),
A tag cannot be used as a value (we're coercing to strings, but conceptually).
